app/src/main/java/com/example/androidacademyhw/movies/FragmentMoviesList.kt

```diff
@@ -41,43 +37,23 @@ class FragmentMoviesList : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        coroutineScope.launch(Dispatchers.Main) {
-            val movies = getMovies(requireContext())
-            initAdapter(movies)
-        }
+        viewModel.movies.observe(viewLifecycleOwner, { movies -> moviesAdapter.submitList(movies) })
+        initAdapter()
     }
 
     override fun onDestroy() {
         super.onDestroy()
         _binding = null
     }
 
-    private fun initAdapter(movies: List<Movie>) {
-        moviesAdapter = MoviesAdapter(
-                object : MoviesAdapter.OnMovieClickListener {
-                    override fun onMovieClick(movie: Movie) {
-                        onMovieDetailsClicked(movie)
-                    }
-                }
-            )
-
-        moviesAdapter.apply {
-            setHasStableIds(true)
-            submitList(movies)
-        }
+    private fun initAdapter() {
+        moviesAdapter = MoviesAdapter(this)
         binding.listMovies.apply {
-            setHasFixedSize(true)
             adapter = moviesAdapter
```

Comment:
in the onDestroyView u should nullify the adapter to avoid memory leak

Reply:
moviesAdapter is lateinit. Is there any options to nullify it? And by nullify you mean setting it to null?

Reply:
`recyclerView.adapter = null`

Reply:
Done